rules/consistent-test-it.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+const getNodeName = require('./util').getNodeName;
+const isTestCase = require('./util').isTestCase;
+const isDescribe = require('./util').isDescribe;
+
+module.exports = {
+  meta: {
+    docs: {
+      url:
+        'https://github.com/jest-community/eslint-plugin-jest/blob/master/docs/rules/consistent-test-it.md',
+    },
+    fixable: 'code',
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          fn: {
+            enum: ['it', 'test'],
+          },
+          withinDescribe: {
+            enum: ['it', 'test'],
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+  create(context) {
+    const configObj = context.options[0] || {};
+    const testKeyword = configObj.fn || 'test';
+    const testKeywordWithinDescribe =
+      configObj.withinDescribe || configObj.fn || 'it';
+
```

Comment:
This top part is missing some coverage, mind adding the necessary tests to cover all cases?

![image](https://user-images.githubusercontent.com/1404810/36075722-2448d4a4-0f53-11e8-82b7-bb77f6e3e68d.png)